### DISCUSSION TITLE
Hoover Washer/Dryer machine support: H-WASH 300 Pro HBDOS695TAMCET

### DISCUSSION THREAD

**Discussion by geekypenguin** [ofalvai/home-assistant-candy#286] [Device support improvements]
Decoded JSON:
`        "statusLavatrice":{
                "WiFiStatus":"0",
                "Err":"255",
                "MachMd":"2",
                "Pr":"5",
                "PrPh":"2",
                "SLevel":"255",
                "Temp":"40",
                "SpinSp":"16",
                "Opt1":"0",
                "Opt2":"0",
                "Opt3":"0",
                "Opt4":"0",
                "Opt5":"0",
                "Opt6":"0",
                "Opt7":"0",
                "Opt8":"0",
                "Steam":"0",
                "DryT":"0",
                "DelVal":"255",
                "RemTime":"49",
                "RecipeId":"0",
                "CheckUpState":"0"
        }
`

What I have decoded so far:
McMd -> Machine Mode? 1 when ready, 2 when running
Pr -> Program. 1-15 (Full program list below)
PrPH = Washing Phase
SLevel -> Stain Level 1,2,3 or 255 for off
Temp -> Washing temperature, min 20 max 90
SpinSp -> Spin Speed /100 (16=1600, 14=1400 etc). min 0, max 16.
Opt1-8 -> Additional options. 1 = Prewash, 8 = Aqua plus / Extra Rinse. Rest unknown.
Steam -> Steam program 1-3
DryT -> Dry Type. 0=off, 1=Extra Dry, 2=Cupboard dry, 3=Iron Dry, 4=Cooldown, 5=120 mins,  6=90 mins, 7=60 mins, 8=30 mins
DelVal -> Delay Start in hours, 1-24, 255 for off
RemTime -> Remaining time in mins
RecipeId -> Unknown, possibly related to sequences that can be built with simplifi app
CheckUpState -> Unknown, possibly binary for if the checkup program is due.

Programs:

**Comment by geekypenguin** [comment]
Programs
1 Dry High
2 Dry Low
3 Dry Wool
4 Extra Care
5 All in One 59
6 Rapid Care 14/30/44
7 Active Steam
8 Rinse
9 Spin/ Drain
10 Mixed
11 20degree Eco
12 Wool & Soft Care
13 Eco 40-60
14 Cottons
15 OneFI

**Comment by ademathews-afk** [comment]
Hi, firstly thanks very much for maintaining this integration - it’s really useful to have local access to these Candy/Hoover appliances in Home Assistant.

I’m using a Hoover H-WASH 300 Pro and have noticed what looks like an issue with the washing-machine remaining-time calculation.

During a running cycle, the washing machine displays e.g. 1:55 remaining, while Home Assistant showed 2 minutes remaining.

Looking at the current WashingMachineStatus.from_json() code, I can see:

`remaining_minutes=round(int(json["RemTime"]) / 60)`

This appears to treat RemTime as seconds, but on my machine it looks to already be reported in minutes. So a value of around 115 becomes round(115 / 60) = 2.

For comparison, the tumble-dryer and dishwasher parsers appear to use int(json["RemTime"]) directly.

Other values from the machine are working correctly - current state, programme phase, temperature, spin speed, etc.

More than happy to provide a decoded status response / diagnostics if that would help confirm the behaviour. Thanks again for the work on the integration.
